client/my-sites/plugins/marketplace/marketplace-plugin-setup-status/index.tsx

```diff
@@ -1,53 +1,108 @@
-}: {
-	pluginSlug?: string;
-} ): JSX.Element {
+function WrappedMarketplacePluginSetup(): JSX.Element {
+	const [ currentStage, setCurrentStage ] = useState( 1 );
+	const [ simulatedProgressPercentage, setSimulatedProgressPercentage ] = useState( 0 );
+	const stages = [ 'Installing plugin', 'Activating plugin' ];
+	const interval = 2000;
+
+	const dispatch = useDispatch();
 	const selectedSite = useSelector( getSelectedSite );
+	const transferStatus = useSelector( ( state ) =>
+		getAutomatedTransferStatus( state, selectedSite?.ID )
+	);
+	const { pluginSlugToBeInstalled, isPluginInstalledDuringPurchase } = useSelector(
+		getPurchaseFlowState
+	);
+
+	useEffect( () => {
+		if ( pluginSlugToBeInstalled ) {
+			dispatch( initiateThemeTransfer( selectedSite.ID, null, pluginSlugToBeInstalled ) );
+		} else if ( isPluginInstalledDuringPurchase ) {
+			alert( 'To be implemented: polling wait to be implemented to check for transfer' );
+		} else {
+			// Invalid State redirect to Yoast marketplace page for now, and maybe a marketplace home view in the future
+			page(
+				`/marketplace/product/details/wordpress-seo${
+					selectedSite?.slug ? `/${ selectedSite.slug }` : ''
+				}`
+			);
+		}
+	}, [ dispatch, pluginSlugToBeInstalled ] );
+
+	useEffect( () => {
+		setTimeout(
+			() => {
+				if ( simulatedProgressPercentage < 95 ) {
+					setSimulatedProgressPercentage( ( percentage ) => percentage + 6 );
+				} else if ( simulatedProgressPercentage !== 100 && simulatedProgressPercentage !== 95 ) {
+					setSimulatedProgressPercentage( 95 );
+				}
+			},
+
+			interval
+		);
+	}, [ simulatedProgressPercentage ] );
 
-	//Plugin status can be extracted with this selector once installation is triggered
-	const pluginStatus = useSelector( ( state ) => {
-		return selectedSite ? getStatusForPlugin( state, selectedSite?.ID, pluginSlug ) : 'UNKNOWN';
-	} );
+	useEffect( () => {
+		if ( transferStatus === 'complete' ) {
+			// TODO: Make sure the primary domain is set to the relevant domain before redireting to thank-you page
+			setSimulatedProgressPercentage( 100 );
+			setTimeout( () => {
+				page(
+					`/marketplace/thank-you${
+						selectedSite?.slug ? `/${ selectedSite.slug }` : ''
+					}?flags=marketplace-yoast`
+				);
+			}, 4000 );
+		}
+	}, [ selectedSite.slug, transferStatus ] );
 
+	if ( simulatedProgressPercentage > 50 && currentStage === 1 ) {
+		setCurrentStage( 2 );
+	}
 	return (
 		<>
 			{ selectedSite?.ID ? <QueryJetpackPlugins siteIds={ [ selectedSite.ID ] } /> : '' }
 			<Masterbar></Masterbar>
 			<div className="marketplace-plugin-setup-status__root">
 				<div>
 					<h1 className="marketplace-plugin-setup-status__title wp-brand-font">
-						Installing Plugin { pluginStatus }
+						{ <div>{ stages[ currentStage - 1 ] }</div> }
 					</h1>
-					<StyledProgressBar value={ 25 } color={ '#C9356E' } compact />
-					<div>Step 1 of 3</div>
+					<StyledProgressBar value={ simulatedProgressPercentage } color="#C9356E" compact />
```

Comment:
Is there a reason that the color couldn't be defined together in the `StyledProgressBar` definition above?

Reply:
Yeah apparently it takes a color prop internally and sets it. Since there is already a prop provided by `StyledProgressBar` to do it, I didn't try it with css.
I am wondering whether we need to include this color in built in that component as well. Happy to do that change if required :)

CC - @razvanpapadopol